playground.sh

```diff
@@ -18,44 +18,107 @@
     exit 1
   fi
 }
 
-testK8s() {
-  echo "Testing K8s environment ..."
-  kubectl cluster-info
-  if [ $? -eq 0 ]; then
-    echo "INFO: K8s is working correctly!"
+checkDockerCompose() {
+  isExist=$(which docker compose)
+
+  if [ ${isExist} ]; then
+    echo "[INFO] Docker compose check passed: Docker compose is working correctly!"
   else
-    echo "ERROR: There was an issue running kubectl cluster-info, please check you K8s cluster."
+    echo "[ERROR] Docker compose check failed: No docker service environment found. Please install docker compose."
     exit 1
   fi
 }
 
+checkDockerDisk() {
+  # Step 1: Get Docker Root Directory
+  local dockerRootDir="$(docker info 2>/dev/null | grep "Docker Root Dir" | awk '{print $NF}')"
 
-checkCompose() {
-  isExist=$(which docker-compose)
-  if [ $isExist ]; then
-    true # Placeholder, do nothing
+  # Step 2: Check if the Docker directory exists
+  if [ -z "${dockerRootDir}" ]; then
+    echo "[ERROR] Docker disk check failed: Docker is not running or Docker Root Directory not found."
+    exit 1
+  fi
+
+  local availableSpaceKB
+
+  if [ -d "${dockerRootDir}" ]; then
+    # Check available space in the Docker directory's partition
+    availableSpaceKB=$(df --output=avail "${dockerRootDir}" | awk 'NR==2 {print $1}')
   else
-    echo "ERROR: No docker service environment found. Please install docker-compose."
-    exit
+    # Check available space in the root partition if the directory doesn't exist (special case for WSL)
+    availableSpaceKB=$(df --output=avail / | awk 'NR==2 {print $1}')
+  fi
+
+  # Step 3: Check if available space is greater than required
+  local availableSpaceGB=$((${availableSpaceKB} / 1024 / 1024))
+
+  if [ "${availableSpaceGB}" -ge "${requiredDiskSpaceGB}" ]; then
+    echo "[INFO] Docker disk check passed: ${availableSpaceGB} GB available."
+  else
+    echo "[ERROR] Docker disk check failed: ${availableSpaceGB} GB available, ${requiredDiskSpaceGB} GB or more required."
+    exit 1
+  fi
+}
+
+checkDockerRam() {
+  local totalRamBytes=$(docker info --format '{{.MemTotal}}')
+  # Convert from bytes to GB
+  local totalRamGB=$((totalRamBytes / 1024 / 1024 / 1024))
+
+  if [ "${totalRamGB}" -ge "${requiredRamGB}" ]; then
+    echo "[INFO] Docker RAM check passed: ${totalRamGB} GB available."
+  else
+    echo "[ERROR] Docker RAM check failed: Only ${totalRamGB} GB available, ${requiredRamGB} GB or more required."
+    exit 1
+  fi
+}
+
+checkDockerCpu() {
+  local cpuCores=$(docker info --format '{{.NCPU}}')
+
+  if [ "${cpuCores}" -ge "${requiredCpuCores}" ]; then
+    echo "[INFO] Docker CPU check passed: ${cpuCores} cores available."
+  else
+    echo "[ERROR] Docker CPU check failed: Only ${cpuCores} cores available, ${requiredCpuCores} cores or more required."
+    exit 1
+  fi
+}
+
+testK8s() {
+  echo "[INFO] Testing K8s environment..."
+  kubectl cluster-info
```

Comment:
I think we need to check if there exists `kubectl` first.

Reply:
Before the `start`, `status`, or `stop` commands are run, the `checkCurrentRuntime` function will be executed.

The `checkCurrentRuntime` function checks whether `kubectl` is present.

![image](https://github.com/user-attachments/assets/5082adde-a885-4f99-b7ce-38babf2baef5)

![image](https://github.com/user-attachments/assets/2310dbb8-6bb1-4787-8320-7954d0676443)